src/components/MyKiva/JournalUpdatesCarousel.vue

```diff
@@ -127,11 +142,27 @@ const singleSlideWidth = computed(() => {
 	return '422px';
 });
 
-const openLightbox = updateId => {
+const getCheckoutReceipt = async () => {
+	const update = updates.value.find(u => u.id === clickedUpdate.value);
+	const response = await apollo.query({
+		query: thanksPageReceiptQuery,
+		variables: {
+			checkoutId: update.id,
+			visitorId: cookieStore.get('uiv') || null,
+		}
+	});
+
+	receipt.value = response?.data?.shop?.receipt ?? null;
+};
+
+const openLightbox = async updateId => {
 	clickedUpdate.value = updateId;
 	const update = props.updates.find(u => u.id === updateId);
 	updateSubject.value = update.subject;
 	updateBody.value = update.body;
+	if (update?.isTransaction) {
+		await getCheckoutReceipt();
```

Comment:
How much of delay in opening the modal is this? Do we need to rather show the modal and have a loader?